tooltip.lua

```diff
@@ -9,10 +9,21 @@ function ns.dataobj.OnTooltipShow(tip)
 	if num then
 		for i=1,num do
 			local specID, spec = GetSpecializationInfo(i)
-			tip:AddLine(" ")
-			tip:AddLine(spec)
-			for j,stat in pairs(ns.priorities[specID]) do
-				tip:AddLine(stat, 1,1,1)
+			local priorities = ns.priorities[specID]
+			if type(priorities[1]) ~= "table" then
+				priorities = {priorities}
```

Comment:
This is a source of garbage chrun and could be bad if our tooltip is being updated frequently (it shouldn't be, but tooltips have a long history of abuse).  Especially when you consider that the source data is static, we're allocating a new table every time for something that doesn't change.

I'd break the bit that adds the details to the tooltip out into its own function, and pass it a table and title from this function so that you're not allocating new tables.